### PR TITLE
docs: adding hivemind

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

adding ingo about HivemindStrategy, https://github.com/Lightning-AI/lightning-Hivemind

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda

### CODE REVIEW

**Review by akihironitta — APPROVED** [review]

File: requirements/pytorch/strategies.txt [hunk 1]
```diff
@@ -2,4 +2,5 @@
 #  in case you want to preserve/enforce restrictions on the latest compatible version, add "strict" as an in-line comment
 
 deepspeed>=0.6.0, <0.8.0  # TODO: Include 0.8.x after https://github.com/microsoft/DeepSpeed/commit/b587c7e85470329ac25df7c7c2521ff9b2833db7 gets released
-lightning-colossalai==0.1.0dev1
+lightning-colossalai>=0.1.0dev1
+lightning-hivemind>=0.1.0dev0
```
Comment (awaelchli) [inline requirements/pytorch/strategies.txt:6]:
```suggestion
```
We don't need to install this one in our CI. There aren't any tests with it, right?

File: docs/source-pytorch/advanced/third_party/hivemind.rst [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+Under the hood, we use `Hivemind <https://github.com/learning-at-home/hivemind>`_ which provides de-centralized training across the internet.
+
+.. warning::  This is an :ref:`experimental <versioning:Experimental API>` feature.
+
+
+To use Collaborative Training, you need to first this extension.
+
+.. code-block:: bash
+
+    pip install lightning-Hivemind
```
Comment (awaelchli) [inline docs/source-pytorch/advanced/third_party/hivemind.rst:19]: why is the package capitalized? Shouldn't it be `pip install lightning-hivemind`?

Comment (Borda) [reply docs/source-pytorch/advanced/third_party/hivemind.rst:19]:
```suggestion
    pip install lightning-hivemind
```

**Review by awaelchli — APPROVED** [review]

File: docs/source-pytorch/extensions/strategy.rst [hunk 3]
```diff
@@ -116,8 +116,13 @@ There are powerful third-party strategies that integrate well with Lightning but
    * - colossalai
      - `Lightning-AI/lightning-colossalai <https://github.com/Lightning-AI/lightning-colossalai>`_
      - Colossal-AI provides a collection of parallel components for you. It aims to support you to write your distributed deep learning models just like how you write your model on your laptop. `Learn more. <https://www.colossalai.org/>`__
+   * - hivemind
+     - `Lightning-AI/lightning-hivemind <https://github.com/Lightning-AI/lightning-hivemind>`_
+     - Hivemind is a PyTorch library for decentralized deep learning across the Internet. Its intended usage is training one large model on hundreds of computers from different universities, companies, and volunteers. `Learn more. <https://github.com/learning-at-home/hivemind>`__
 
 
+training on unreliable mixed GPUs across the internet
```
Comment (carmocca) [inline docs/source-pytorch/extensions/strategy.rst:124]: Is this text meant to be here? Looks cut off

File: docs/source-pytorch/extensions/strategy.rst [hunk 4]
```diff
@@ -116,8 +116,13 @@ There are powerful third-party strategies that integrate well with Lightning but
...
+
```
Comment (Borda) [inline docs/source-pytorch/extensions/strategy.rst:125]:
```suggestion
```

**Pull request merged**